RubeesEat/Model/InMemoryBillRepository.cs

```diff
@@ -26,6 +26,39 @@ public IReadOnlyList<Bill> GetAll()
             : null;
     }
 
+    private BalanceChange GetBalanceChangesForUser(Bill bill, Person currentUser)
+    {
+        var sum = bill.EntryLines
+                      .Where(entryLine => entryLine.Person.Id == currentUser.Id)
+                      .Sum(entryLine => entryLine.Amount);
+        return new BalanceChange(sum, bill.Date, bill.Description, bill.Id);
+    }
+
+    public List<BalanceChange> GetBalanceChangesForPage(Person user, int? pageIndex, int? pageSize, int? amount)
+    {
+        var balanceChanges = _bills.Values
+                                   .Where(b => b.EntryLines.Any(e => e.Person.Id == user.Id))
+                                   .OrderByDescending(b => b.Date);
+        if (amount.HasValue)
+        {
+            return balanceChanges.Take((int)amount)
+                                 .Select(b => GetBalanceChangesForUser(b, user))
+                                 .ToList();
+        }
+
+        if (pageIndex.HasValue && pageSize.HasValue)
+        {
+            var skip = (pageIndex - 1) * pageSize;
+
+            return balanceChanges.Skip((int)skip)
+                                 .Take((int)pageSize)
+                                 .Select(b => GetBalanceChangesForUser(b, user))
+                                 .ToList();
+        }
+
+        return new List<BalanceChange>();
+    }
+
```

Comment:
Maybe there is a way to simplify this. 

1. We can use our helper function to reduce duplication
2. Currently, this code gets all balance changes from a user, then makes another list out of that, based on if we want a certain amount or balance changes for the Show Bills page.

We can simplify by removing the amount parameter. Instead, if we want recent balance changes, we set pageIndex to 1 and pageSize to the amount of pages we want. 

So we can leave this function as we had it previously